### ISSUE TITLE
Transfer learning with Resnet50 fail with Exception

### ISSUE THREAD

**Issue by MrXu** [fchollet/deep-learning-models#24]
Hi, I am using Resnet50 to do transfer learning. The backend is tensorflow.
I tried to stack three more layers on top of the Resnet but fail with following error:
```
Exception: The shape of the input to "Flatten" is not fully defined (got (None, None, 2048). 
Make sure to pass a complete "input_shape" or "batch_input_shape" argument to the first layer in your model.
```

The code for stacking two models are as following:
```
    model = ResNet50(include_top=False, weights='imagenet')

    top_model = Sequential()
    top_model.add(Flatten(input_shape=model.output_shape[1:]))
    top_model.add(Dense(256, activation='relu'))
    top_model.add(Dropout(0.5))
    top_model.add(Dense(1, activation='sigmoid'))
    top_model.load_weights(top_model_weights_path)

    model = Model(input=model.input, output=top_model(model.output))
```

**Comment by AntreasAntoniou** [comment]
You should use the functional API. So for example try:

`    if K.image_dim_ordering() == 'tf':
        inp = Input(shape=(224, 224, 3), name='input_image')
    else:
        inp = Input(shape=(3, 224, 224), name='input_image')

    
    main_model = VGG16()(inp)
    
    main_out = Dropout(0.5)(main_model)
    main_out = Dense(512, activation='relu', name='class_id')(main_out)
    main_out = Dropout(0.5)(main_model)
    main_out = Dense(output_classes, activation='softmax', name='class_id')(main_out)

    model = Model(input=inp, output=artist_out)`

**Comment by MrXu** [comment]
@AntreasAntoniou Thanks, but I need to load weights for the Sequential model. Also, when I try to use `main_model = Resnet50(include_top=False)(inp)`, it complains `Exception: You are attempting to share a same `BatchNormalization` layer across different data flows.`

**Comment by MrXu** [comment]
I solved the problem after reading through source code for `Resnet50.py`. If anyone encounter the same problem, please add the `input_shape` parameter when you call `Resnet50()`:
```
model = ResNet50(include_top=False, weights='imagenet', input_tensor=Input(shape=(224,224,3)))
```

If you are using Keras `ImageDataGenerator`, the input shape will be automatically inferred.

**Comment by alyato** [comment]
@MrXu 
hi, i also use the resnet50, and i want set `input_size=(48,48,3)` but  it show me an error.
>valueerror input size must be at least 197x197,

i check the doc, it show me ,

> width and height should be no smaller than 32. E.g. (200, 200, 3) would be one valid value.,

It's so weird.

**Comment by alonalj** [comment]
Regarding the error:
> input size must be at least 197x197,

Had the same issue, but disappeared after upgrading keras to 2.2.4 (with backend tensorflow 1.12.0 in case it matters).
